filebeat/input/filestream/copytruncate_prospector.go

```diff
@@ -239,79 +239,90 @@ func (p *copyTruncateFileProspector) Run(ctx input.Context, s loginp.StateMetada
 				return nil
 			}
 
-			src := p.identifier.GetSource(fe)
-			log = loggerWithEvent(log, fe, src)
+			p.onFSEvent(log, ctx, fe, s, hg, ignoreInactiveSince)
 
-			switch fe.Op {
-			case loginp.OpCreate, loginp.OpWrite:
-				if fe.Op == loginp.OpCreate {
-					log.Debugf("A new file %s has been found", fe.NewPath)
+		}
+		return nil
+	})
 
-				} else if fe.Op == loginp.OpWrite {
-					log.Debugf("File %s has been updated", fe.NewPath)
-				}
+	errs := tg.Wait()
+	if len(errs) > 0 {
+		log.Error("%s", sderr.WrapAll(errs, "running prospector failed"))
+	}
+}
 
-				if p.fileProspector.isFileIgnored(log, fe, ignoreInactiveSince) {
-					continue
-				}
+func (p *copyTruncateFileProspector) onFSEvent(
+	log *logp.Logger,
+	ctx input.Context,
+	fe loginp.FSEvent,
+	s loginp.StateMetadataUpdater,
+	hg loginp.HarvesterGroup,
```

Comment:
maybe you want to rename some of the parameters like `event`, `updater`, `group` ?